sim/pokemon.ts

```diff
@@ -525,6 +525,35 @@ export class Pokemon {
 		return null;
 	}
 
+	getMoveHitData(move: ActiveMove) {
+		move = Pokemon.ensureActiveMove(move);
+		const slotid = this.toString().slice(0, 3);
+		return {
+			crit: move.moveHitData.crit[slotid] || false,
+			typeMod: move.moveHitData.typeMod[slotid] || 0,
+			zBrokeProtect: move.moveHitData.zBrokeProtect[slotid] || false,
+		};
```

Comment:
I'm wondering whether the move hit data should be stored the other way around, i.e. `move.moveHitData[slotid][datatype]` instead of `move.moveHitData[datatype][slotid]`. I guess it would look something like this:

    getMoveHitData(move: ActiveMove) {
        const slotid = this.toString().slice(0, 3);
        if (!move.moveHitData) move.moveHitData = {};
        if (!move.moveHitData[slotid]) move.moveHitData[slotid] = { crit: false, typeMode: 0, zBrokeProtect: false };
        return move.moveHitData[slotid];
    }

    setMoveCrit(move: ActiveMove) {
        this.getMoveHitData(move).crit = true;
    }

    setMoveTypeModFor(move: ActiveMove, typeMod: number) {
        this.getMoveHitData(move).typeMod = number;
    }

    setMoveZBreakProtect(move: ActiveMove) {
        this.getMoveHitData(move).zBrokeProtect = true;
    }

Reply:
I actually did bring this up earlier. I agree, that would be better.

Reply:
I'll work on this myself.

Reply:
Ah, sorry, I failed to make the connection.